spec/chrome_integration/gtalk.spec.js

```diff
@@ -0,0 +1,237 @@
+        '&grant_type=refresh_token';
+    ajax('POST', 'https://www.googleapis.com/oauth2/v3/token', data)
+        .then(function(resp) {
+          fulfill(resp.access_token);
+        });
+  });
+}
+
+// Returns a promise that fulfills when socialClient has been logged in
+// using the loginEmail address.  loginEmail must have an associated
+// refresh token.
+function loginAs(socialClient, loginEmail) {
+  var refreshToken = REFRESH_TOKENS[loginEmail.toLowerCase()];
+  if (!refreshToken) {
+    return Promise.reject('No refresh token found for ' + loginEmail);
+  } else if (pendingAccessToken) {
+    return Promise.reject('loginAs cannot be called concurrently.' +
+        '  Please wait until previous login completes');
+  }
+  return getAccessToken(refreshToken).then(function(accessToken) {
+    // Set the accessToken to pendingAccessToken so that it can be used by
+    // our OAuthView.
+    pendingAccessToken = accessToken;
+    return socialClient.login({
+        agent: 'integration',
+        version: '0.1',
+        url: '',
+        interactive: false,
+        rememberLogin: false
+      });
+  });
+}
+
+// Sets up an onClientState listener and fulfills the returned Promise
+// when the given userId appears as ONLINE.
+function waitForUser(socialClient, userId) {
+  return new Promise(function(fulfill, reject) {
+    socialClient.on('onClientState', function(clientState) {
+      if (clientState.userId == userId &&
+          clientState.status == 'ONLINE') {
+        fulfill(clientState);
+      }
+    });
+  });
+}
+
+describe('GTalk', function() {
+  var socialInterface;
+
+  // Message to be sent between peers.  If a unique message is not used,
+  // messages from one persons test might interfere with another person who
+  // is running the same tests at the same time.
+  var uniqueMsg = Math.random().toString();
+
+  function logoutThenDone(socialClientsArray, done) {
+    var promises = [];
+    for (var i = 0; i < socialClientsArray.length; ++i) {
+      promises.push(socialClientsArray[i].logout());
+    }
+    Promise.all(promises).then(done);
+  }
+
+  beforeEach(function(done) {
+    if (!socialInterface) {
+      freedom('scripts/dist/social.google.json',
+          {oauth: [OAuthView], debug: 'log'})
+          .then(function(interface) {
+        // Store socialInterface so we don't have to reload freedom
+        // before each test.
+        socialInterface = interface;
+        done();
+      }.bind(this));
+    } else {
+      done();
+    }
+  }.bind(this));
+
+  it('Can login and logout', function(done) {
+    var socialClient = socialInterface();
+    loginAs(socialClient, ALICE_EMAIL).then(function(clientInfo) {
+      expect(clientInfo.userId).toEqual(ALICE_EMAIL);
+      expect(clientInfo.clientId).toEqual(ALICE_EMAIL + '/integration');
+      expect(clientInfo.status).toEqual('ONLINE');
+      logoutThenDone([socialClient], done);
+    });
+  });
+
+  it('Can send messages', function(done) {
```

Comment:
I think the 'Can send messages' and 'Can send 8 messages per second' test cases are fine without expect statements.  Adding an expect that the messageData.message == uniqueMsg is redundant to the check we already have - and I think its better not to add that expect statement for all messages since there is a small chance bobSocialClient will receive a message from somewhere outside this test